test/gtest/uct/ib/test_ib_md.cc

```diff
@@ -155,4 +162,105 @@ UCS_TEST_P(test_ib_md, aligned) {
     ib_md_umr_check(&rkey_buffer[0], true, size, true);
 }
 
+ucs_status_t test_ib_md::reg_pack(void *buffer, size_t size,
+                                  uct_ib_mem_t **memh, uct_rkey_t *rkey_p)
+{
+    std::string rkey_buffer(md_attr().rkey_packed_size, '\0');
+    uct_rkey_bundle_t bundle;
+
+    ucs_status_t status = uct_md_mem_reg(md(), buffer, size,
+                                         UCT_MD_MEM_ACCESS_REMOTE_ATOMIC |
+                                                 UCT_MD_MEM_SYMMETRIC_RKEY,
+                                         (void**)memh);
+    if (status == UCS_OK) {
+        status = uct_md_mkey_pack(md(), *memh, &rkey_buffer[0]);
+        if ((status == UCS_OK) && (rkey_p != NULL)) {
+            status = uct_rkey_unpack(md()->component, &rkey_buffer[0], &bundle);
+            if (status == UCS_OK) {
+                *rkey_p = bundle.rkey;
+                status  = uct_rkey_release(md()->component, &bundle);
+            }
+        }
+    }
+
+    return status;
+}
+
+// Symmetric keys properties for the generic IB transport
+void test_ib_md::check_smkeys(uct_rkey_t rkey1, uct_rkey_t rkey2)
+{
+    EXPECT_NE(UCT_IB_INVALID_MKEY, uct_ib_md_atomic_rkey(rkey1));
+    EXPECT_NE(UCT_IB_INVALID_MKEY, uct_ib_md_direct_rkey(rkey1));
+    EXPECT_NE(UCT_IB_INVALID_MKEY, uct_ib_md_atomic_rkey(rkey2));
+    EXPECT_NE(UCT_IB_INVALID_MKEY, uct_ib_md_direct_rkey(rkey2));
+    EXPECT_EQ(uct_ib_md_atomic_rkey(rkey1) - uct_ib_md_direct_rkey(rkey1),
+              uct_ib_md_atomic_rkey(rkey2) - uct_ib_md_direct_rkey(rkey2));
+    EXPECT_EQ(uct_ib_md_direct_rkey(rkey1) + 0x100,
```

Comment:
where is `0x100` dfefined?

Reply:
it's +1 shifted by 8 bits tag width.

Reply:
1. Can we somehow check if keys are really created in a symmetric way?
2. What if 2 or more processes will run this test on the same NIC at the same time?

Reply:
1. difficult in gtest, specially if many are running the same test in parallel. If we were to implement a test under `test/apps`, we'd need to run a single instance, or introduce sub-ranges concept.

2. works because we check atomic/rkey offset and only allocate 2-3 keys in one acquired block(< 8 keys/block), so they will always be contiguous (+0x100), confirmed it with two gtest repeating

Reply:
maybe run the test 2 times one after another (after destroying and re-creating the MD) and check we get the same index?
we can add such test in next PR, but here at least need to remove line 198